Description: PCSD web interface listens only on IPv6
 The problem seems to be in the change to `WEBrick::Utils::create_listeners`
 introduced in ruby 2.1:
 .
 https://github.com/ruby/ruby/commit/b1f493dcd1092fe17cccec998e175516ed5c6e47
 .
 A small example program calling this function can prove this:
   WEBrick::Utils::create_listeners("::", 2000)
 .
 This is the strace output for CentOS7 with ruby 2.0:
   socket(PF_INET6, SOCK_STREAM|SOCK_CLOEXEC, IPPROTO_TCP) = 7
   fcntl(7, F_GETFD)                       = 0x1 (flags FD_CLOEXEC)
   fstat(7, {st_mode=S_IFSOCK|0777, st_size=0, ...}) = 0
   fstat(7, {st_mode=S_IFSOCK|0777, st_size=0, ...}) = 0
   setsockopt(7, SOL_SOCKET, SO_REUSEADDR, [1], 4) = 0
   bind(7, {sa_family=AF_INET6, sin6_port=htons(2000), inet_pton(AF_INET6, "::", &sin6_addr), sin6_flowinfo=0, sin6_scope_id=0}, 28) = 0
   listen(7, 128)                          = 0
 .
 On ruby 2.1 and newer the same program executes this:
   socket(PF_INET6, SOCK_STREAM|SOCK_CLOEXEC, IPPROTO_TCP) = 7
   fcntl(7, F_GETFD)                       = 0x1 (flags FD_CLOEXEC)
   fstat(7, {st_mode=S_IFSOCK|0777, st_size=0, ...}) = 0
   fstat(7, {st_mode=S_IFSOCK|0777, st_size=0, ...}) = 0
   fstat(7, {st_mode=S_IFSOCK|0777, st_size=0, ...}) = 0
   fstat(7, {st_mode=S_IFSOCK|0777, st_size=0, ...}) = 0
   getsockname(7, {sa_family=AF_INET6, sin6_port=htons(0), inet_pton(AF_INET6, "::", &sin6_addr), sin6_flowinfo=0, sin6_scope_id=0}, [28]) = 0
   setsockopt(7, SOL_IPV6, IPV6_V6ONLY, [1], 4) = 0
   getsockname(7, {sa_family=AF_INET6, sin6_port=htons(0), inet_pton(AF_INET6, "::", &sin6_addr), sin6_flowinfo=0, sin6_scope_id=0}, [28]) = 0
   setsockopt(7, SOL_SOCKET, SO_REUSEADDR, [1], 4) = 0
   bind(7, {sa_family=AF_INET6, sin6_port=htons(2000), inet_pton(AF_INET6, "::", &sin6_addr), sin6_flowinfo=0, sin6_scope_id=0}, 28) = 0
   listen(7, 128)                          = 0
 .
 The problem is caused by the IPV6_V6ONLY socket option being set:
   setsockopt(7, SOL_IPV6, IPV6_V6ONLY, [1], 4) = 0
 .
 On ruby 2.1 using ``nil`` in place of ``::`` has the same effect so
 this ruby version check is added to the bind code in pcsd.
Author: Valentin Vidic <Valentin.Vidic@CARNet.hr>
Bug: https://github.com/ClusterLabs/pcs/issues/51
Last-Update: 2016-06-17
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/pcsd/ssl.rb
+++ b/pcsd/ssl.rb
@@ -105,6 +105,10 @@
   end
 end
 
+if primary_addr == '::' and RUBY_VERSION >= '2.1'
+  primary_addr = nil
+end
+
 webrick_options = {
   :Port               => 2224,
   :BindAddress        => primary_addr,
